trlx/trlx.py

```diff
@@ -89,8 +89,12 @@ def train(  # noqa: C901
             eval_prompts = prompts[:batch_size]
 
         pipeline = get_pipeline(config.train.pipeline)(prompts, max_prompt_length, trainer.tokenizer)
-        orch = get_orchestrator(config.train.orchestrator)(trainer, pipeline, chunk_size=config.method.chunk_size)
-        orch.make_experience(config.method.num_rollouts)
+        trainer.add_prompt_pipeline(pipeline)
```

Comment:
Re `add_prompt_pipeline` - yeah that is a bit awkward, it should be possible to pass it via args if you move the `get_trainer` call into the PPO part of the branch?
You could also probably replace the `get_pipeline` with `PromptPipeline` since all the models use the same pipeline

Reply:
* Yeah I wasn't sure if it'd be too messy - let me give it a go :)

* I agree on replacing the `get_pipeline` stuff with just `PromptPipeline`; I originally did that but reverted before creating the PR to limit the scope to the issue being addressed. I can't think of any other prompt pipelines so not sure what the `_DATAPIPELINE` registry is even intended for?

Reply:
> so not sure what the _DATAPIPELINE registry is even intended for?
It's from the time when every pipeline was ought to be a specific dataset, each registered deliberately